common/src/main/java/org/apache/drill/common/util/GuavaPatcher.java

```diff
@@ -24,154 +24,157 @@
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 
-import javassist.CannotCompileException;
 import javassist.ClassPool;
 import javassist.CtClass;
 import javassist.CtConstructor;
 import javassist.CtMethod;
 import javassist.CtNewMethod;
-import javassist.NotFoundException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class GuavaPatcher {
   private static final Logger logger = LoggerFactory.getLogger(GuavaPatcher.class);
 
-  private static boolean patched;
+  private static boolean patchingAttempted;
 
   public static synchronized void patch() {
-    if (!patched) {
-      try {
-        patchStopwatch();
-        patchCloseables();
-        patchPreconditions();
-        patched = true;
-      } catch (Throwable e) {
-        logger.warn("Unable to patch Guava classes.", e);
-      }
+    if (!patchingAttempted) {
+      patchingAttempted = true;
```

Comment:
This version no longer has a try/catch block. Is the idea that, if patching fails, the test should fail? If this particular test fails, we've marked patching as attempted and we won't try again. This means that future tests that needed the patching will mysteriously fail.

Should we, instead, simply do a `System.exit(1)` if patching fails to shut down the tests? Or, print a warning to stderr that subsequent tests may fail?

Reply:
The try/catch block was moved to the component methods. It is simpler this way to pinpoint possible errors.

Reply:
For the case when patching is used when starting Drill, it would be better to avoid killing a drillbit if it will fail. In this case, part of the functionality would still be available.
Regarding unit tests use case, there was introduced a test that checks that patching is produced for all test classes.